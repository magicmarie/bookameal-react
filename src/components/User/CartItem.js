import React from "react";

// cart item card: customer side
const CartItem = props => (
  <div>
    <tr>
      <th scope="row">{props.mealName}</th>
      <td>{props.price}</td>
      <td>3</td>
      <td>{props.adminName}</td>
      <td>30000UGX</td>
      <td>
        <button
          className="fa fa-check-circle"
          onClick={() => props.handleAddOrder(props.id, props.mealId)}
        />
      </td>
      <td>
        <button className="fa fa-remove" />
      </td>
    </tr>
  </div>
);
export default CartItem;
