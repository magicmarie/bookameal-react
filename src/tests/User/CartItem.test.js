import React from "react";
import { shallow } from "enzyme";
import CartItem from "../../components/User/CartItem";

describe("UserDashboard component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CartItem removeItem={jest.fn()} onChange={jest.fn()} />);
  });
  it("renders successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("renders successfully", () => {
    const handleEvent = jest.fn();
    wrapper.find("button.fa-remove").simulate("click", handleEvent);
  });
  it("renders successfully", () => {
    const handleEvent = jest.fn();
    wrapper.find('[name="quantity"]').simulate("change", handleEvent);
  });
});
