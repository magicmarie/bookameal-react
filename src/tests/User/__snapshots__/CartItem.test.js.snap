// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDashboard component renders successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CartItem
    onChange={[Function]}
    removeItem={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td />,
        <td />,
        <td>
          <select
                    defaultValue={undefined}
                    name="quantity"
                    onChange={[Function]}
          >
                    <option
                              value="1"
                    >
                              1
                    </option>
                    <option
                              value="2"
                    >
                              2
                    </option>
                    <option
                              value="3"
                    >
                              3
                    </option>
                    <option
                              value="4"
                    >
                              4
                    </option>
                    <option
                              value="5"
                    >
                              5
                    </option>
                    <option
                              value="6"
                    >
                              6
                    </option>
                    <option
                              value="7"
                    >
                              7
                    </option>
                    <option
                              value="8"
                    >
                              8
                    </option>
                    <option
                              value="9"
                    >
                              9
                    </option>
                    <option
                              value="10"
                    >
                              10
                    </option>
          </select>
</td>,
        <td>
          UGX 
</td>,
        <td>
          UGX 
          NaN
</td>,
        <td>
          <button
                    className="fa fa-remove text-danger"
                    onClick={[Function]}
          />
</td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <select
            defaultValue={undefined}
            name="quantity"
            onChange={[Function]}
>
            <option
                        value="1"
            >
                        1
            </option>
            <option
                        value="2"
            >
                        2
            </option>
            <option
                        value="3"
            >
                        3
            </option>
            <option
                        value="4"
            >
                        4
            </option>
            <option
                        value="5"
            >
                        5
            </option>
            <option
                        value="6"
            >
                        6
            </option>
            <option
                        value="7"
            >
                        7
            </option>
            <option
                        value="8"
            >
                        8
            </option>
            <option
                        value="9"
            >
                        9
            </option>
            <option
                        value="10"
            >
                        10
            </option>
</select>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="1"
>
                1
</option>,
              <option
                value="2"
>
                2
</option>,
              <option
                value="3"
>
                3
</option>,
              <option
                value="4"
>
                4
</option>,
              <option
                value="5"
>
                5
</option>,
              <option
                value="6"
>
                6
</option>,
              <option
                value="7"
>
                7
</option>,
              <option
                value="8"
>
                8
</option>,
              <option
                value="9"
>
                9
</option>,
              <option
                value="10"
>
                10
</option>,
            ],
            "defaultValue": undefined,
            "name": "quantity",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1",
                "value": "1",
              },
              "ref": null,
              "rendered": "1",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2",
                "value": "2",
              },
              "ref": null,
              "rendered": "2",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3",
                "value": "3",
              },
              "ref": null,
              "rendered": "3",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4",
                "value": "4",
              },
              "ref": null,
              "rendered": "4",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5",
                "value": "5",
              },
              "ref": null,
              "rendered": "5",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "6",
                "value": "6",
              },
              "ref": null,
              "rendered": "6",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "7",
                "value": "7",
              },
              "ref": null,
              "rendered": "7",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "8",
                "value": "8",
              },
              "ref": null,
              "rendered": "8",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "9",
                "value": "9",
              },
              "ref": null,
              "rendered": "9",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "10",
                "value": "10",
              },
              "ref": null,
              "rendered": "10",
              "type": "option",
            },
          ],
          "type": "select",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "UGX ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "UGX ",
          undefined,
        ],
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "UGX ",
            NaN,
          ],
        },
        "ref": null,
        "rendered": Array [
          "UGX ",
          NaN,
        ],
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="fa fa-remove text-danger"
            onClick={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-remove text-danger",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td />,
          <td />,
          <td>
            <select
                        defaultValue={undefined}
                        name="quantity"
                        onChange={[Function]}
            >
                        <option
                                    value="1"
                        >
                                    1
                        </option>
                        <option
                                    value="2"
                        >
                                    2
                        </option>
                        <option
                                    value="3"
                        >
                                    3
                        </option>
                        <option
                                    value="4"
                        >
                                    4
                        </option>
                        <option
                                    value="5"
                        >
                                    5
                        </option>
                        <option
                                    value="6"
                        >
                                    6
                        </option>
                        <option
                                    value="7"
                        >
                                    7
                        </option>
                        <option
                                    value="8"
                        >
                                    8
                        </option>
                        <option
                                    value="9"
                        >
                                    9
                        </option>
                        <option
                                    value="10"
                        >
                                    10
                        </option>
            </select>
</td>,
          <td>
            UGX 
</td>,
          <td>
            UGX 
            NaN
</td>,
          <td>
            <button
                        className="fa fa-remove text-danger"
                        onClick={[Function]}
            />
</td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <select
              defaultValue={undefined}
              name="quantity"
              onChange={[Function]}
>
              <option
                            value="1"
              >
                            1
              </option>
              <option
                            value="2"
              >
                            2
              </option>
              <option
                            value="3"
              >
                            3
              </option>
              <option
                            value="4"
              >
                            4
              </option>
              <option
                            value="5"
              >
                            5
              </option>
              <option
                            value="6"
              >
                            6
              </option>
              <option
                            value="7"
              >
                            7
              </option>
              <option
                            value="8"
              >
                            8
              </option>
              <option
                            value="9"
              >
                            9
              </option>
              <option
                            value="10"
              >
                            10
              </option>
</select>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="1"
>
                  1
</option>,
                <option
                  value="2"
>
                  2
</option>,
                <option
                  value="3"
>
                  3
</option>,
                <option
                  value="4"
>
                  4
</option>,
                <option
                  value="5"
>
                  5
</option>,
                <option
                  value="6"
>
                  6
</option>,
                <option
                  value="7"
>
                  7
</option>,
                <option
                  value="8"
>
                  8
</option>,
                <option
                  value="9"
>
                  9
</option>,
                <option
                  value="10"
>
                  10
</option>,
              ],
              "defaultValue": undefined,
              "name": "quantity",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "value": "1",
                },
                "ref": null,
                "rendered": "1",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2",
                  "value": "2",
                },
                "ref": null,
                "rendered": "2",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "3",
                  "value": "3",
                },
                "ref": null,
                "rendered": "3",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "4",
                  "value": "4",
                },
                "ref": null,
                "rendered": "4",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "5",
                  "value": "5",
                },
                "ref": null,
                "rendered": "5",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "6",
                  "value": "6",
                },
                "ref": null,
                "rendered": "6",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "7",
                  "value": "7",
                },
                "ref": null,
                "rendered": "7",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "8",
                  "value": "8",
                },
                "ref": null,
                "rendered": "8",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "9",
                  "value": "9",
                },
                "ref": null,
                "rendered": "9",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "10",
                  "value": "10",
                },
                "ref": null,
                "rendered": "10",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "UGX ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "UGX ",
            undefined,
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "UGX ",
              NaN,
            ],
          },
          "ref": null,
          "rendered": Array [
            "UGX ",
            NaN,
          ],
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="fa fa-remove text-danger"
              onClick={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-remove text-danger",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "button",
          },
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
